name.abuchen.portfolio/src/name/abuchen/portfolio/model/Client.java

```diff
@@ -187,6 +188,53 @@ public void removeSecurity(final Security security)
 
         propertyChangeSupport.firePropertyChange("securities", security, null); //$NON-NLS-1$
     }
+    
+    /**
+     * Gets a list of used {@link CurrencyUnit}s.
+     * 
+     * @return list
+     */
+    public List<CurrencyUnit> getUsedCurrencies()
+    {
+        // collect all used currency codes
+        HashSet<String> hsUsedCodes = new HashSet<String>();
+        // first client and all accounts
+        hsUsedCodes.add(baseCurrency);
+        for (Account account : accounts)
+        {
+            hsUsedCodes.add(account.getCurrencyCode());
+            for (AccountTransaction t : account.getTransactions())
```

Comment:
Die ```AccountTransaction``` darf keine andere Währung haben als das Konto selber. Den Loop über die Buchungen kann man sich mMn sparen.

Reply:
Tauchen denn zum Beispiel Fremdwährungsdividenden dann als Euro in der `AccountTransaction` auf? Ansonsten nehme ich diesen Teil heraus.

Reply:
@buchen ist jetzt geändert, er nimmt nur noch den `Account` an sich.